confluence_utils/cli.py

```diff
@@ -1,39 +1,188 @@
+        elif parent_file.page_id is None:
+            # check if the page exist but the page id is not set
+            if confluence.page_exists(space=space, title=parent_file.title):
+                parent_id = confluence.get_page_id(space, parent_file.title)
+                parent_file.page_id = parent_id
+                markdown_file.parent_id = parent_id
+                markdown_file.update_front_matter(space)
+                parent_file.update_front_matter(space)
+            else:
+                click.echo(f"parent page does not exist skipping: {path}")
+                return
+
+    body, attachments = markdown_file.render_confluence_content()
+    page_with_title_exist = confluence.page_exists(
+        space=space, title=markdown_file.title
+    )
+
+    if not markdown_file.page_id and page_with_title_exist:
+        page_id = confluence.get_page_id(space, markdown_file.title)
+        markdown_file.page_id = page_id
+        markdown_file.update_front_matter(space)
+        update_page_response = confluence.update_page(
+            page_id=markdown_file.page_id,
+            title=markdown_file.title,
+            body=body,
+            parent_id=markdown_file.parent_id,
+        )
+        click.echo(
+            f"Updated page with page_id {update_page_response.get('id')}"
+        )
+    elif markdown_file.page_id and page_with_title_exist:
+        # look up page id to make sure confluence is aware of it
+        checkedSpace = None
+        try:
+            checkedSpace = confluence.get_page_space(markdown_file.page_id)
+        except ApiError:
+            pass
+        if checkedSpace is None or checkedSpace != space:
+            # the page exist but the page id is bung so fix the page id
+            page_id = confluence.get_page_id(space, markdown_file.title)
+            markdown_file.page_id = page_id
+            markdown_file.update_front_matter(space)
+
+        update_page_response = confluence.update_page(
+            page_id=markdown_file.page_id,
+            title=markdown_file.title,
+            body=body,
+            parent_id=markdown_file.parent_id,
+        )
+
+        click.echo(
+            f"Updated page with page_id {update_page_response.get('id')}"
+        )
+    else:
+        create_page_response = confluence.create_page(
+            space=space,
+            title=markdown_file.title,
+            body=body,
+            parent_id=markdown_file.parent_id,
+        )
+        page_id = create_page_response.get("id")
+        markdown_file.page_id = page_id
+        markdown_file.update_front_matter(space)
+
+    for attachment in attachments:
+        attachment_absolution_path = os.path.join(
+            markdown_file.directory_name, attachment
+        )
+        attachment_filename = os.path.basename(attachment_absolution_path)
+
+        confluence.attach_file(
+            filename=attachment_absolution_path,
+            name=attachment_filename,
+            page_id=page_id,
```

Comment:
@ttrout55 I think it's possible for `page_id` to be undefined here.

Reply:
in order for page_id to be undefined here create or update page would have to return a non 200 in those circumstances the confulence api would raise and exception so we would error out before this point in time. Do you still want to handle this?